packages/services/f-braze-adapter/src/services/utils/_tests/locationFilter.test.js

```diff
@@ -43,135 +49,55 @@ const MOCK_CARDS_WITHOUT_LOCATION_KVP = [
-                longitude: '0.000'
-            });
-            const card = filteredCards.find(c => c.id === '3');
-
+            const filteredCards = locationFilter(MOCK_CARDS_WITH_LOCATION_KVP, locationObject);
             // Assert
-            expect(card).toEqual(undefined);
-            expect(filteredCards.length).toEqual(2);
-        });
-
-        it('should return cards that have latitude and longitude PLACEHOLDERS where the user has an invalid latitude and longitude and they DO have a location attribute', () => {
-            // Arrange & Act
-            const filteredCards = locationFilter(MOCK_CARDS_WITH_LOCATION_KVP, {
-                location: 'BA11 2PT',
-                latitude: '0.000',
-                longitude: '0.000'
-            });
-
-            // Assert
-            expect(filteredCards.length).toEqual(4);
-        });
-
-        it('should return cards that do NOT have location PLACEHOLDERS', () => {
-            // Arrange & Act
-            const filteredCards = locationFilter(MOCK_CARDS_WITH_LOCATION_KVP, {
-                location: null,
-                latitude: null,
-                longitude: null
-            });
-
-            // Assert
-            expect(filteredCards.length).toEqual(2);
-        });
-
-        it('should return cards where the users location matches the cards location KVP', () => {
-            // Arrange & Act
-            const filteredCards = locationFilter(MOCK_CARDS_WITH_LOCATION_KVP, MOCK_CURRENT_LOCATION);
-            const card = filteredCards.find(c => c.id === '4');
-            const card2 = filteredCards.find(c => c.id === '5');
-
-            // Assert
-            expect(card).toEqual(MOCK_CARDS_WITH_LOCATION_KVP[3]);
-            expect(card2).toEqual(undefined);
-            expect(filteredCards.length).toEqual(4);
+            expect(filteredCards.length).toEqual(numberOfCards);
         });
+    });
 
-        it('should NOT return cards where the users location does NOT match the cards location KVP', () => {
+    describe('card has NO Location KVP', () => {
+        it.each([
+            [4, { location: MOCK_CURRENT_LOCATION.location, latitude: MOCK_CURRENT_LOCATION.latitude, longitude: MOCK_CURRENT_LOCATION.longitude }],
+            [3, { location: null, latitude: MOCK_CURRENT_LOCATION.latitude, longitude: MOCK_CURRENT_LOCATION.longitude }],
+            [3, { location: MOCK_CURRENT_LOCATION.location, latitude: null, longitude: null }],
+            [3, { location: MOCK_CURRENT_LOCATION.location, latitude: '0.000', longitude: '0.000' }],
+            [3, { location: MOCK_CURRENT_LOCATION.location, latitude: null, longitude: MOCK_CURRENT_LOCATION.longitude }],
+            [3, { location: MOCK_CURRENT_LOCATION.location, latitude: MOCK_CURRENT_LOCATION.latitude, longitude: null }],
+            [1, { location: null, latitude: null, longitude: null }]
+        ])('should return %s cards when the users location is %o', (numberOfCards, locationObject) => {
```

Comment:
For this and the similar section above, I think for understandability's sake it would be good to have a simple description string describing each case, and to help identifying the test case in question:
```suggestion
            [4, "has all location attributes", { location: MOCK_CURRENT_LOCATION.location, latitude: MOCK_CURRENT_LOCATION.latitude, longitude: MOCK_CURRENT_LOCATION.longitude }],
            [3, "only has geocoords", { location: null, latitude: MOCK_CURRENT_LOCATION.latitude, longitude: MOCK_CURRENT_LOCATION.longitude }],
            [3, "only has location", { location: MOCK_CURRENT_LOCATION.location, latitude: null, longitude: null }],
            [3, "has invalid geocoords", { location: MOCK_CURRENT_LOCATION.location, latitude: '0.000', longitude: '0.000' }],
            [3, "is missing latitude", { location: MOCK_CURRENT_LOCATION.location, latitude: null, longitude: MOCK_CURRENT_LOCATION.longitude }],
            [3, "is missing longitude", { location: MOCK_CURRENT_LOCATION.location, latitude: MOCK_CURRENT_LOCATION.latitude, longitude: null }],
            [1, "has no valid location attributes", { location: null, latitude: null, longitude: null }]
        ])('should return %s cards when the users location %s', (numberOfCards, description, locationObject) => {
```